build: add teamcity scripts for test coverage

RaduBerinde opened the pull request:
#### cli: add package comment

Fix for this error saw in a bazel coverage run:
```
pkg/cli/keytype_string.go:3:1: at least one file in a package should have a package comment (ST1000)
```

Epic: none
Release note: None

#### bazci: fix null pointer access

I ran into a null pointer access on `summary`. Some of the existing
code does check for `summary != nil` so it is an expected possible
condition. This fix moves the use of the field behind a check.

Epic: none
Release note: None

#### build: add teamcity scripts for test coverage

This commit adds teamcity scripts for generating unit test coverage.

We include some small modifications to `bazci` that allow using
`coverage` and saving the `coverage.dat` files to the artifacts.

Epic: none
Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/110225)
<!-- Reviewable:end -->

RaduBerinde commented:
Sample build: https://teamcity.cockroachdb.com/viewLog.html?buildId=11708297&tab=buildResultsDiv&buildTypeId=Cockroach_Coverage_UnitTests&branch_Cockroach_Coverage=110225

healthy-pod reviewed (commented):
We try to let scripts path in the repository map to TeamCity's project path. In TeamCity, this lives under `Cockroach/Coverage` so we need to create a new directory `build/teamcity/cockroach/coverage` and then move the scripts to `build/teamcity/cockroach/coverage`.

Follow-ups as a result of this:
- Updating `dir="$(dirname $(dirname $(dirname $(dirname $(dirname "${0}")))))"` to make 4 `dirname` calls only
- Updating the script path in the build config setting in TeamCity

RaduBerinde commented:
Oh, that makes sense, thanks! Done, I will update the build and make sure it still works.

RaduBerinde committed:
cli: add package comment

Fix for this error saw in a bazel coverage run:
```
pkg/cli/keytype_string.go:3:1: at least one file in a package should have a package comment (ST1000)
```

Epic: none
Release note: None

RaduBerinde committed:
build: add teamcity scripts for test coverage

This commit adds teamcity scripts for generating unit test coverage.

We include some small modifications to `bazci` that allow using
`coverage` and saving the `coverage.dat` files to the artifacts.

Epic: none
Release note: None

RaduBerinde committed:
bazci: fix null pointer access

I ran into a null pointer access on `summary`. Some of the existing
code does check for `summary != nil` so it is an expected possible
condition. This fix moves the use of the field behind a check.

Epic: none
Release note: None

RaduBerinde commented:
Renamed the scripts (they no longer need `coverage` in the name) and added CCL versions.

healthy-pod reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

RaduBerinde commented:
TFTR!

bors r+
